### ISSUE TITLE
Write Iteration with suspicious time only to debug log

### ISSUE THREAD

**Issue by pauroger** [catboost/catboost#514]
Problem: {Iterations with suspicious time}
catboost version: {version}
Operating System: {OS}
CPU: {CPU}

This is my attempt at applying BayesSearch in CatBoost:

from catboost import CatBoostClassifier
from skopt import BayesSearchCV
from sklearn.model_selection import StratifiedKFold

# Classifier
  bayes_cv_tuner = BayesSearchCV(
      estimator = CatBoostClassifier(
          silent=True
      ),
    search_spaces = {
        'depth':(2,16),
        'l2_leaf_reg':(1, 500),
        'bagging_temperature':(1e-9, 1000, 'log-uniform'),
        'border_count':(1,255),
        'rsm':(0.01, 1.0, 'uniform'),
        'random_strength':(1e-9, 10, 'log-uniform'),
        'scale_pos_weight':(0.01, 1.0, 'uniform'),
    },    
    scoring = 'roc_auc',
    cv = StratifiedKFold(
        n_splits=2,
        shuffle=True,
        random_state=72
    ),
    n_jobs = 1,
    n_iter = 100,   
    verbose = 1,
    refit = True,
    random_state = 72
)

# 
def status_print(optim_result):
    """Status callback durring bayesian hyperparameter search"""

    # Get all the models tested so far in DataFrame format
    all_models = pd.DataFrame(bayes_cv_tuner.cv_results_)    
    
    # Get current parameters and the best parameters    
    best_params = pd.Series(bayes_cv_tuner.best_params_)
    print('Model #{}\nBest ROC-AUC: {}\nBest params: {}\n'.format(
        len(all_models),
        np.round(bayes_cv_tuner.best_score_, 4),
        bayes_cv_tuner.best_params_
    ))

# Fit the model
resultCAT = bayes_cv_tuner.fit(X_train, y_train, callback=status_print)

# Output
Fitting 2 folds for each of 1 candidates, totalling 2 fits
[Parallel(n_jobs=1)]: Done   2 out of   2 | elapsed:  1.3min finished
Model #1
Best ROC-AUC: 0.9078
Best params: {'bagging_temperature': 7.385095136875193e-08, 'border_count': 235, 'depth': 6, 'l2_leaf_reg': 87, 'random_strength': 2.374264057108563e-05, 'rsm': 0.6732603131901663, 'scale_pos_weight': 0.2513036640642842}

Fitting 2 folds for each of 1 candidates, totalling 2 fits
[Parallel(n_jobs=1)]: Done   2 out of   2 | elapsed:  5.6min finished
Model #2
Best ROC-AUC: 0.9078
Best params: {'bagging_temperature': 7.385095136875193e-08, 'border_count': 235, 'depth': 6, 'l2_leaf_reg': 87, 'random_strength': 2.374264057108563e-05, 'rsm': 0.6732603131901663, 'scale_pos_weight': 0.2513036640642842}

Fitting 2 folds for each of 1 candidates, totalling 2 fits
[Parallel(n_jobs=1)]: Done   2 out of   2 | elapsed:  1.5min finished
Model #3
Best ROC-AUC: 0.9099
Best params: {'bagging_temperature': 6.356757750213118, 'border_count': 154, 'depth': 7, 'l2_leaf_reg': 392, 'random_strength': 1.323858457287431, 'rsm': 0.7411135466632092, 'scale_pos_weight': 0.01782909461419627}

Fitting 2 folds for each of 1 candidates, totalling 2 fits

Iteration with suspicious time 7.55 sec ignored in overall statistics.

Iteration with suspicious time 739 sec ignored in overall statistics.

(...) and more of the same.

# Request

It would be awesome if you were to post a tutorial on how to properly do this. In the meantime, any ideas of where I went wrong?

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
Such message means that building some tree was unexpectedly fast or slow, and was not added to the performance profile.
You may see such messages, if the overall CPU load varies considerably.

The learning process itself is not affected in this case.

**Comment by annaveronika** [comment]
The code is there, the next pypi version will contain it. Thanks again for the report!
